lib/srv/exec.go

```diff
@@ -365,36 +367,61 @@ func collectStatus(cmd *exec.Cmd, err error) (*execResult, error) {
 }
 
 // getDefaultEnvPath returns the default value of PATH environment variable for
-// new logins (prior to shell)
-//
-// Normally getDefaultEnvPath is set to "" (default /etc/login.defs is used)
-// but for unit testing it takes any file
-//
-// Returns a strings which looks like "PATH=/usr/bin:/bin"
-func getDefaultEnvPath(loginDefsPath string) string {
-	defaultValue := "PATH=" + defaultPath
-	if loginDefsPath == "" {
-		loginDefsPath = "/etc/login.defs"
+// new logins (prior to shell) based on login.defs. Returns a strings which
+// looks like "PATH=/usr/bin:/bin"
+func getDefaultEnvPath(uid string, loginDefsPath string) string {
```

Comment:
This function has gotten way more complex... can I challenge you to simplify it? Why have the inner func, for example? It's basically an if-expression...